### PR TITLE
Wallet Balance GUI

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Rewrite the wallet balance GUI in vue

## Testing

- Check that the balance updated correctly and displays the correct value in the selected currency
- Check that the GUI updates when a new currency is selected
- Check that the three dots menu items work and display correctly (i.e. Hardware wallets shouldn't have an export key item, legacy wallets don't have a refresh address item)

### CODE REVIEW

**Review by panleone — CHANGES_REQUESTED** [review]
There are still some places where `getBalance(true);` is called but the signal `balance-update` is not emitted

File: scripts/wallet.js [hunk 1]
```diff
@@ -404,7 +402,6 @@ export async function importWallet({
         }
 
         // For non-HD wallets: hide the 'new address' button, since these are essentially single-address MPW wallets
-        if (!wallet.isHD()) doms.domNewAddress.style.display = 'none';
 
         // Update the loaded address in the Dashboard
         wallet.getNewAddress({ updateGUI: true });
```
Comment (panleone) [inline scripts/wallet.js:407]: my fault here on my refactor PR: it was meant to call the global function getNewAddress not the function getNewAddress of the wallet class

**Review by panleone — CHANGES_REQUESTED** [review]

File: scripts/WalletBalance.vue [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    guiRenderCurrentReceiveModal,
+    guiRenderContacts,
+} from './contacts-book';
+import { wallet, getNewAddress } from './wallet';
+
+const balance = ref(0);
+const price = ref(0.0);
+const updating = ref(false);
+const currency = ref('USD');
+const balanceStr = computed(() => {
```
Comment (panleone) [inline scripts/WalletBalance.vue:29]: This value is never updated

File: scripts/global.js [hunk 3]
```diff
@@ -317,6 +306,10 @@ export async function start() {
         domAdvancedModeToggler: document.getElementById('advancedModeToggler'),
     };
 
+    for (const d in doms) {
+        if (!doms[d]) console.log(d);
```
Comment (panleone) [inline scripts/global.js:310]: remove this

**Review by panleone — COMMENTED** [review]
A couple more comments

File: scripts/WalletBalance.vue (hunk 2 above)
Comment (Duddino) [reply scripts/WalletBalance.vue:29]: balanceStr should updated through Vue reactivity

Comment (Duddino) [reply scripts/WalletBalance.vue:30]: Fixed, nDisplayDecimals was not reactive (Also had a typo on settings.js)

File: scripts/settings.js [hunk 4]
```diff
@@ -544,7 +544,8 @@ export async function toggleTestnet() {
     }
 
     mempool.UTXOs = [];
-    getBalance(true);
+
+    getEventEmitter().emit('new-balance');
```
Comment (panleone) [inline scripts/settings.js:548]: I think this should be `balance-update`

**Comment by panleone** [comment]
needs merge or rebase with master

**Comment by Duddino** [comment]
Done

File: scripts/global.js [hunk 5]
```diff
@@ -1824,7 +1797,7 @@ function addProposalToFinalisationCache(cProposal) {
 }
 
 /**
- * Render Governance proposal objects to a given Proposal category
+ * Render Governance proosal objects to a given Proposal category
```
Comment (JSKitty) [inline scripts/global.js:1800]: Someone stole a **P**

```suggestion
 * Render Governance proposal objects to a given Proposal category
```

**Review by JSKitty — CHANGES_REQUESTED** [review]
A couple of small display issues and regressions spotted, nothing major.
- The 'Refresh' button does nothing when clicked, functional change or regression.
- There's no space between the Ticker or Currency now, which looks odd compared to `Master`.
- All 'mini-menu' icons also have no spaces before the button texts, looks odd.

# PR / Master

<div style=\"display: flex; justify-content: center;\">
  pr
  master
</div>

**Comment by Duddino** [comment]
All done

**Review by JSKitty — APPROVED** [review]
**tACK** 62efebb1303934527da88427920ec6130c36d52e

Pretty clean implementation, functionally/visually identical to `Master`.

**Review by panleone — APPROVED** [review]
tACK 62efebb1303934527da88427920ec6130c36d52e good work

**Pull request merged**